### DISCUSSION TITLE
O que vocês utilizam para teste de acessibilidade e testes de SEO?

### DISCUSSION THREAD

**Discussion by mvfsillva** [frontendbr/forum#66] [Perguntas e Respostas] [answered]

**Comment by bernardodiasc** [comment, marked as answer]
Olá, tenho alguns links anotados sobre esse assunto, espero que sejam úteis. O validador Unicorn do W3C é o principal, digamos assim.

### Análise e validação
- http://validator.w3.org - W3C Markup Validation Service (HTML and XHTML, SVG 
  or MathM ... )
- http://validator.w3.org/unicorn - Unicorn - Validador Unificado do W3C
- http://jigsaw.w3.org/css-validator/ - CSS Validation Service
- http://manifest-validator.com/ - Cache Manifest Validator
- http://validator.w3.org/checklink - Link Checker W3C

### Análise de Semântica
- http://www.w3.org/2003/12/semantic-extractor.html - Semantic Data Extractor,
  W3C
- http://validator.w3.org/feed/ - Feed Validation Service, W3C
- http://www.w3.org/RDF/Validator/rdfval - RDF Validation Service, W3C
- http://www.google.com/webmasters/tools/richsnippets - Dados estruturados, Google
- https://www.google.com/webmasters/markup-tester - Markup Tester para e-mail,
  Google
- https://webmaster.yandex.com/microtest.xml - Structured data validator, Yandex
  (microdata, schema.org, microformats, OpenGraph and RDF)
- https://gsnedders.html5.org/outliner/ - Document Outline extractor

### Análise de Acessibilidade
- http://www.w3.org/WAI/ER/tools/
- http://www.cynthiasays.com/
- http://wave.webaim.org/ - WAVE Web Accessibility Tool
- http://achecker.ca/checker/index.php - Web Accessibility Checker
- http://www.contentquality.com - HiSoftware® Cynthia Says™ , (Section 508, WCAG 2.0)
- http://www.dasilva.org.br - Da Silva, Governo Brasileiro
- http://www.acessibilidade.gov.pt/accessmonitor/ - AccessMonitor

**Comment by brunopulis** [comment]
@mvfsilva para acessibilidade recomendo e reforço o que o @bernardodiasc disse, use o AccessMonitor e o WAVE que você vai conseguir bons resultados já.

**Comment by mvfsillva** [comment]
@bernardodiasc e @brunopulis Obrigado!

**Comment by artdiniz** [comment]
Muito boas as dicas do @bernardodiasc! Pra acessibilidade eu adicionaria um item: aprender a usar bem algum leitor de tela e testar seus sites nele. Se tiver um Mac o [VoiceOver](http://webaim.org/articles/voiceover/) já é bem legal. No Windows dá pra usar o [NVDA](http://webaim.org/articles/nvda/) que é gratuito, ou o [JAWS](http://webaim.org/articles/jaws/) (o mais usado) em sessões de 40min. Não conheço nenhum de linux =o

**Comment by brunopulis** [comment]
@artdiniz Existe pra Linux o [Orca ](https://wiki.gnome.org/Projects/Orca) e um muito hardocore o [Lynx](http://lynx.browser.org/).

Na minha opinião todo frontend deveria aprender a usar um leitor de tela para compreender as barreiras que muitas vezes criamos por desconhecimento de uma parcela enorme de usuários que utilizam essas tecnologias assistivas. 

E vale lembrar que o NVDA é mais compatível com o Firefox :)

**Comment by brunopulis** [comment]
Segue os manuais de como utilizar os leitores citados pelo o @artdiniz 

[Manual do NVDA](http://www.acessibilidadelegal.com/33-nvda.php)
[Manual do Orca](http://www.acessibilidadelegal.com/33-manual-orca.php)

**Comment by vitor-mendrone-deel** [comment]
Estou iniciando testes com essa ferramenta para testes de acessibilidade: 

[Pa11y](http://pa11y.org/)

A mágica está no fato de que você pode implementar direto no servidor pra que ele faça uma monitoria constante em diversos sites. 

Também uso essa lib JS

[Ally.js](http://allyjs.io/)

**Comment by brunopulis** [comment]
@Mendrone Muito legal vou colocar no repositório do [Awesome A11Y](https://github.com/brunopulis/awesome-a11y)

**Comment by vitor-mendrone-deel** [comment]
@brunopulis Que legal! :smile: 

Tem essa aqui também, um pouco mais simples, em CSS: 

[Diagnostic.css](http://www.karlgroves.com/2013/09/07/diagnostic-css-super-quick-web-accessibility-testing/)

**Comment by felipefialho** [comment]
Interessante essa heim...

**Comment by brunopulis** [comment]
@Mendrone muito bacana o Karl Groves sempre arrebentando! :+1:

**Comment by AdrianoCahete** [comment]
Só dando aquele upzinho, o Google lançou faz uns dias o [Chromelens](https://chrome.google.com/webstore/detail/chromelens/idikgljglpfilbhaboonnpnnincjhjkd).

É uma extensão que mostra como algumas pessoas com deficiência visual enxergam o seu site, desde Daltonismo até cegueira total. Faz alguns testes também, mostrando o que tá "errado".

![image1](https://lh3.googleusercontent.com/IIcKnxXpjZ9l7CgWN2obH8Wmw_rpWGp9918o-zLpyZrvE9L3msUT7otq9LSIvawETvLy3j8ZeaE=s640-h400-e365)

![image2](https://lh3.googleusercontent.com/QiO9B4Bws_HnuE247dwhfhWRAFayD69jOEji2ZQvpcoBKaF6q7u0KgUgQ7SvGO-X2JCIybS0eZY=s640-h400-e365)

![image3](https://lh3.googleusercontent.com/Nt-4yQz0EPAYcEFRlMVQV5wK0IR4Ko08aDIf5zevihz3h4LjyPhy4aXGsyt9JOb6_1DsgMeD1A=s640-h400-e365)

Tem, obviamente, no Github também: https://github.com/chromelens/chromelens

**Comment by brunopulis** [comment]
@AdrianoCahete  eu cheguei a ver, achei interessante. Mas me parece estar em versão beta ainda.

**Comment by julianotrovao** [comment]
Bom dia, sou deficiente visual, e trabalho com testes de  acessibilidade, se precisarem de uma consultoria no sistema, estamos aqui. Tanto acessibilidade para sistemas web, e sistemas   mobile. Para sistemas mobile, uma sugestão, é o escanner de  acessibilidade para  android, e, para web, a  extenção e-scaner, também é  interessantíssima para   chrome. Além dos  testes de  acessibilidade que faço, usando leitores de  tela: nvda, orca, virtual vision, jaws,  e nos navegadores: safary (ios),  mozila  e  chrome(ios). Abraços. contato para  consultoria: e-mail: julianotrovao2@gmail.com

**Comment by felipefialho** [comment]
Muito legal @julianotrovao
